runtime/nvqir/QIRTypes.cpp

```diff
@@ -73,30 +107,34 @@ std::vector<int64_t> getRangeValues(Array *in_array, const Range &in_range) {
   // Convert to absolute index.
   const auto start_idx = convertIndex(in_range.start);
   const auto end_idx = convertIndex(in_range.end);
-
+  Array *result = new Array(0, sizeof(Qubit *));
```

Comment:
Do we really always want to allocate memory? It looks like the old algorithm returned a vector of null pointers originally.

Reply:
Actually, the original algorithm returned an empty vector of indices, which will turn into an `Array*` of zero size, like what we have here.

Reply:
Yes. An "empty vector" is 3 pointers initialized to `nullptr`. :)